web/controllers/notes_controller.ex

```diff
@@ -2,6 +2,13 @@ defmodule Notox.NoteController do
   use Notox.Web, :controller
   alias Notox.Note
 
+  def index(conn, %{"search" => %{"query" => query}}) do
+    notes =
+      Note.search(Note, query)
```

Comment:
I would say that the first one is discretionary. The second example I would agree with.